op-node/rollup/derive/system_config.go

```diff
@@ -69,50 +72,88 @@ func ProcessSystemConfigUpdateLogEvent(destSysCfg *eth.SystemConfig, ev *types.L
 	}
 	// indexed 1
 	updateType := ev.Topics[2]
-	// unindexed data
+
+	// Create a reader of the unindexed data
+	reader := bytes.NewReader(ev.Data)
+
+	// Helper function to prevent code duplication.
+	readWord := func() []byte {
+		// Allocate a new slice to return
+		b := make([]byte, 32)
```

Comment:
Would use a `[32]byte` fixed-length return value here as mentioned earlier instead of a slice, but I'll stop nitting, this looks good.

Reply:
Regarding Go: the slice->array trick works and is newer, but the most minimal way is probably to define the return value as named array, and read straight into it

Reply:
And in Go it's fine to return by value instead of by pointer reference, especially for small flat values like a 32 byte array. It should stay on the stack that way too, as it can't really escape if not referenced by pointer.